src/main/java/org/reactome/release/dataexport/configuration/ConfigurationManager.java

```diff
@@ -0,0 +1,257 @@
+
+	private String configFileName;
+	private ConfigurationEntryCollection configurationEntryCollection;
+
+	/**
+	 * Creates a ConfigurationManager object using the default configuration file name of "config.properties".
+	 */
+	public ConfigurationManager() {
+		this.configFileName = DEFAULT_CONFIGURATION_FILE_NAME;
+	}
+
+	/**
+	 * Creates a ConfigurationManager object using the provided configuration file name.
+	 *
+	 * @param configFileName Name of the configuration file name to check and potentially create/overwrite
+	 */
+	ConfigurationManager(String configFileName) {
+		this.configFileName = configFileName;
+	}
+
+	/**
+	 * Obtains and returns the path to the sample configuration file included in this project's resources.
+	 *
+	 * @return The file path to the sample configuration file in this project as a String
+	 */
+	public static String getPathToOriginalSampleConfigurationFile() {
+		return Objects.requireNonNull(
+			ConfigurationManager.class.getClassLoader().getResource("sample_config.properties")
+		).getPath();
+	}
+
+	/**
+	 * Creates a property configuration file with the needed property values for the data exporter project.  The file
+	 * will be (re)created if the "overwriteExistingFile" parameter is true.  Otherwise, the file will only be
+	 * created if the file does not already exist or if the existing file is not valid (i.e. it does not have one or
+	 * more of the needed property values for the project).  If the file is (re)created, the file permissions will be
+	 * set to 660 to allow read and write access to the user and group only.
+	 *
+	 * @param overwriteExistingFile <code>true</code> if the file should be (re)created regardless if it already exists
+	 * or is valid;<code>false</code> otherwise
+	 * @return <code>true</code> if the configuration file is (re)created because the file does not exist, is not
+	 * valid, or was requested to be overwritten;<code>false</code> otherwise (i.e the configuration file already
+	 * exists, is valid, and was not requested to be overwritten)
+	 * @throws IOException  Thrown if unable to read an existing configuration file or if unable to write a new
+	 * configuration file if one needs to be recreated (this includes an exception from deleting an existing file,
+	 * writing a new file, or setting its file system permissions appropriately).
+	 */
+	public boolean createConfigurationFile(boolean overwriteExistingFile) throws IOException {
+		if (!overwriteExistingFile && configurationFileExistsAndIsValid()) {
+			return false;
+		}
+
+		initializeConfigurationEntryCreator();
+
+		addNeo4jDatabaseConfigurationEntries();
+		addReactomeSpecificConfigurationEntries();
+		addEuropePMCFTPServerConfigurationEntries();
+		addNCBIFTPServerConfigurationEntries();
+
+		if (!configurationFileExists()) {
+			System.out.println(
+				"The configuration file " + getConfigFileName() + " does not exist.  "
+					+ "Please provide the following property values.");
```

Comment:
Weird indentation here. Is it supposed to look like this?

Reply:
I updated the indentation and formatting to reduce the number of lines (https://github.com/reactome/release-data-exporter/pull/4/commits/3924632bde7f6db33379c6abf8a0684023076ab8), but it looks weird in GitHub's diff in the commit/PR overview.

It looks a little clearer when looking at the file directly: https://github.com/reactome/release-data-exporter/blob/27c44452d301ec514286ce78acfbf581ebd37463/src/main/java/org/reactome/release/dataexport/configuration/ConfigurationManager.java